src/units/abilities.cpp

```diff
@@ -1120,6 +1120,56 @@ void attack_type::modified_attacks(bool is_backstab, unsigned & min_attacks,
 	}
 }
 
+static void damage_type_list(unit_ability_list& abil_list)
+{	//check if have abilities or specials alignment and remove damage_type bad specified
+	const std::set<std::string> checking_damage = {"arcane", "blade", "cold", "fire", "impact", "pierce"};
```

Comment:
This does not seem sustainable. There shouldn't be anything that requires enumerating all the damage types, because anyone can define a custom damage type.